packages/zoe/src/contractSupport/zoeHelpers.js

```diff
@@ -421,3 +422,66 @@ export async function saveAllIssuers(zcf, issuerKeywordRecord = {}) {
   );
   return Promise.all(issuersPSaved);
 }
+
+/** @type {MapKeywords} */
+export const mapKeywords = (keywordRecord, keywordMapping) => {
+  if (keywordMapping === undefined) {
+    return keywordRecord;
+  }
+  return Object.fromEntries(
+    Object.entries(keywordRecord).map(([keyword, value]) => [
+      keywordMapping[keyword],
+      value,
+    ]),
+  );
+};
+/** @type {Reverse} */
+const reverse = keywordRecord => {
+  if (keywordRecord === undefined) {
+    return keywordRecord;
+  }
+  return Object.fromEntries(
+    Object.entries(keywordRecord).map(([key, value]) => [value, key]),
+  );
+};
+
+/** @type {OfferTo} */
+export const offerTo = async (
+  zcf,
+  invitation,
+  proposal,
+  fromSeat,
+  fromAssets,
+  toSeat,
+  keywordMapping = undefined,
+) => {
+  const zoe = zcf.getZoeService();
+
+  const payments = await withdrawFromSeat(zcf, fromSeat, fromAssets);
+
+  // Map to the other contract's keywords
+  const paymentsForOtherContract = mapKeywords(payments, keywordMapping);
+
+  const userSeatPromise = E(zoe).offer(
+    invitation,
+    proposal,
+    paymentsForOtherContract,
+  );
+
+  const depositedPromiseKit = makePromiseKit();
+
+  E(userSeatPromise)
+    .getPayouts()
+    .then(async payoutPayments => {
+      const amounts = await E(userSeatPromise).getCurrentAllocation();
```

Comment:
This `await` doesn't appear to be top-level. Wouldn't it be possible to do 

```suggestion
Promise.all([E(userSeatPromise).getCurrentAllocation(),  E(userSeatPromise).getPayouts()])
    .then(async ([amounts, payoutPayments] => {

```

Reply:
What's the definition of top-level? This is an anonymous function, but it could easily be declared outside beforehand, where awaits would be "top-level". I think there's something wrong with how we are using the term since these two things seem equivalent to me.